File: mvp/src/main/java/com/thefuntasty/taste/injection/module/TasteModule.java
package com.thefuntasty.taste.injection.module;

import android.content.Context;

import com.thefuntasty.taste.display.TDisplay;
import com.thefuntasty.taste.injection.annotation.qualifier.ApplicationContext;
import com.thefuntasty.taste.res.TRes;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class TasteModule {

	@Provides
	@Singleton static TRes tRes(@ApplicationContext Context context) {
		return new TRes(context);
	}

	@Provides
	@Singleton static TDisplay tDisplay(@ApplicationContext Context context) {
		return new TDisplay(context);
	}
}

File: mvp/src/main/java/com/thefuntasty/taste/injection/module/TasteModule.kt
package com.thefuntasty.taste.injection.module

import android.content.Context
import com.thefuntasty.taste.display.TDisplay
import com.thefuntasty.taste.injection.annotation.ApplicationContext
import com.thefuntasty.taste.res.TRes
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class TasteModule {

	@Provides
	@Singleton
	fun tRes(@ApplicationContext context: Context) = TRes(context)

	@Provides
	@Singleton
	fun tDisplay(@ApplicationContext context: Context) = TDisplay(context)
}
